### PR TITLE
feat: implement delete form from dashboard feature

### LINKED ISSUE #4299: [React] Delete form feature
Missed feature...

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR implements the form deletion flow for form owners. 
Closes #4299 

## Solution

**Breaking Changes** 

- [x] No - this PR is backwards compatible  

**Features**:

- Add delete form modal and feature

## Before & After Screenshots
New DeleteFormModal story

### CODE REVIEW

File: frontend/src/features/workspace/components/DeleteFormModal/DeleteFormModal.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+          <Text color="secondary.500">
+            You will lose all responses and feedback for the following form
+            permanently. Are you sure you want to delete the form?
+          </Text>
+          <OrderedList
+            spacing="0.5rem"
+            listStyleType="none"
+            ml="1.75rem"
+            mt="1rem"
+          >
```
Comment (justynoh) [inline frontend/src/features/workspace/components/DeleteFormModal/DeleteFormModal.tsx]: just a thought: is there a reason why we use `OrderedList` here? seems a bit strange to use it for just one item haha. nts if doing a `Grid` would be more natural but achieve the same effect, nbd tho

**Review by justynoh — APPROVED** [review]
lgtm!

File: frontend/src/features/workspace/components/DeleteFormModal/DeleteFormModal.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import { FormId } from '~shared/types'
+
+import { MOCK_USER, userHandlers } from '~/mocks/msw/handlers/user'
+
+import { DeleteFormModal, DeleteFormModalProps } from './DeleteFormModal'
+
+export default {
+  title: 'Pages/WorkspacePage/DeleteFormModal',
+  component: DeleteFormModal,
+  // decorators: [(storyFn) => <MemoryRouter>{storyFn()}</MemoryRouter>],
```
Comment (wanlingt) [inline frontend/src/features/workspace/components/DeleteFormModal/DeleteFormModal.stories.tsx]: should we delete this and the `MemoryRouter` import?

File: frontend/src/features/workspace/components/DeleteFormModal/DeleteFormModal.tsx (hunk 1 above)
Comment (karrui) [reply frontend/src/features/workspace/components/DeleteFormModal/DeleteFormModal.tsx]: I'll probably change to a normal unordered list, but this is preoptimizing for future bulk delete hehe

File: frontend/src/features/workspace/components/DeleteFormModal/DeleteFormModal.stories.tsx (hunk 2 above)
Comment (karrui) [reply frontend/src/features/workspace/components/DeleteFormModal/DeleteFormModal.stories.tsx]: cheers, f9dfdc2

File: frontend/src/features/workspace/components/DeleteFormModal/DeleteFormModal.tsx (hunk 1 above)
Comment (karrui) [reply frontend/src/features/workspace/components/DeleteFormModal/DeleteFormModal.tsx]: 1602960

**Review by timotheeg — APPROVED** [review]

**Pull request merged**